cvxpy/tests/solver_test_helpers.py

```diff
@@ -130,35 +130,36 @@ def check_complementarity(self, places) -> None:
             
     def check_stationary_lagrangian(self, places) -> None:
         L = self.prob.objective.expr
+        objective = self.prob.objective
+        if objective.NAME == 'minimize':
+            L = objective.expr
+        else:
+            L = -objective.expr
         for con in self.constraints:
             if isinstance(con, (cp.constraints.Inequality,
                                 cp.constraints.Equality)):
                 dual_var_value = con.dual_value
                 prim_var_expr = con.expr
-                if self.prob.objective.NAME == 'minimize':
-                    L = L + cp.scalar_product(dual_var_value, prim_var_expr)
-                else:
-                    L = L - cp.scalar_product(dual_var_value, prim_var_expr)
+                L = L + cp.scalar_product(dual_var_value, prim_var_expr)
             elif isinstance(con, (cp.constraints.ExpCone,
                                   cp.constraints.SOC,
                                   cp.constraints.Zero,
                                   cp.constraints.NonNeg,
                                   cp.constraints.PSD,
                                   cp.constraints.PowCone3D)):
-                if self.prob.objective.NAME == 'minimize':
-                    L = L - cp.scalar_product(con.args, con.dual_value)
-                else:
-                    L = L + cp.scalar_product(con.args, con.dual_value)
+                L = L - cp.scalar_product(con.args, con.dual_value)
             else:
                 raise NotImplementedError()
         g = L.grad
         bad_norms = []
 
-        """The reason 'bad' norms could arise (outside of an error by CVXPY) would be if a
-        constraint was introduced in a variable via a flag during it's declaration and not
-        explicitly in the list of constraints passed to the problem --- thus making the above
-        constructed lagrangian incorrect. In these cases instead of checking if the dLdX = 0
-        we check if dLdX \in K^{*} (i.e. the dual cone of the implicit constraint)"""
+        """The convention that we follow for construting the Lagrangian is: 1) Move all
+        explicitly passed constraints to the problem (via Problem.constraints) into the
+        Lagrangian --- dLdX == 0 for any such variables 2) Constraints that have
+        implicitly been imposed on variables at the time of declaration via specific
+        flags (e.g.: PSD/symmetric etc.), in such a case we check, `dLdX\in K^{*}`, where
+        `K` is the convex cone corresponding to the implicit constraint on `X`
+        """
         for (opt_var, v) in g.items():
             if all(not attr for attr in list(map(lambda x: x[1], opt_var.attributes.items()))):
```

Comment:
This if-statement is impossible to decipher. Define a helper function that computes this value. The helper function can be an instance method of the Variable class or it could be a method you define here, in ``solver_test_helpers``.

Reply:
Alternatively, you can just have this case be captured by the "else" statement in this chain of if-else statements.

Reply:
Understood! I think the latter would be neater (which would look fine even when we eventually add in support for complex variables and have to deal with the `complex,hermitian` flags.